libsql/src/database.rs

```diff
@@ -393,6 +433,21 @@ impl Database {
             _ => unreachable!("no database type set"),
         }
     }
+
+    #[cfg(all(feature = "encryption-at-rest", feature = "replication"))]
+    pub fn set_encryption_key(&mut self, encryption_key: impl AsRef<[u8]>) {
```

Comment:
we should add a doc attribute here for feature flags. You can look at the macro.rs file for how those macros setup the feature flags. We should add those to all public fns where we add additional feature flags not done via the macros (since those add it automatically).

Reply:
(oh and that function is going away btw, I added it early in a futile attempt to not introduce all the additional params, but I ended up needing them anyway)